support/android/apk/servoview/src/main/java/org/mozilla/servoview/Servo.java

```diff
@@ -191,6 +193,10 @@ public void onAnimatingChanged(boolean animating) {
             mRunCallback.inGLThread(() -> mGfxCb.animationStateChanged(animating));
         }
 
+        public boolean onAllowNavigation(String url) {
+            return mClient.onAllowNavigation(url);
```

Comment:
So this works without wrapping the call to `mClient.onAllowNavigation` into a `mRunCallback.inUIThread`? Maybe it's better to wrap it, we are not sure the code in `mClient.onAllowNavigation` will always work properly in the GL thread.

Reply:
> Maybe it's better to wrap it, we are not sure the code in mClient.onAllowNavigation will always work properly in the GL thread.

The problem here is we cannot return value (in this case boolean) from Runnable. We'll need a Callable interface for that. I am not entirely sure how to do that here. Since methods `inGLThread` and `inUIThread` take Runnable as arguments.

 If you can give me a little direction I would love to work towards that. 😄